Automattic/jetpack
# Set VideoPress attachment url to originally uploaded video file

Some compatibility issues have come up with regards to VideoPress transcoded videos since the HLS streaming change was introduced. Video players from core blocks or 3rd party plugins were not able to play the HLS playlist. This change is intended to better support those players by returning the originally uploaded video file (still served from the WordPress.com CDN), instead of one of the transcoded video files which are no longer playable as-is in players that don't support HLS streaming.

The only compatibility issues that may remain as a result of this change are if the originally uploaded video file is in a format not widely supported by browsers. In these cases, the customer will need to convert the video file to a more common format (ex h264) with a tool and settings that [we now recommend in support page](https://wordpress.com/support/videopress/recommended-video-settings/).

Fixes # 994-gh-Automattic/greenhouse

#### Changes proposed in this Pull Request:

* The video file returned by the `wp_get_attachment_url()` function used to be a transcoded video file (and since 10.2 it was the m3u8 file, if available), it will now be the originally uploaded video file.

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Visit a Jetpack site with VideoPress enabled
* Upload an h264 video file (either in the Media Library or in a post) (ex [Big Buck Bunny clip](https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/1080/Big_Buck_Bunny_1080_10s_30MB.mp4))
* Add the video to a Cover block on a post, the video should play
* View the "File URL" field in the Media Library for that video, it should be the originally uploaded file name (not end in `_hd.mp4` or `_hd.master.m3u8` - unless that is what your file was named)
